src/main/resources/static/theme-toggle.js

```diff
@@ -1,34 +1,38 @@
 (function () {
   const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)')
-  let initialTheme
-  if (localStorage.getItem('darkmode-pref-set') === 'true') {
-    initialTheme = localStorage.getItem('darkMode') === 'true'
-  } else {
-    initialTheme = darkModeMediaQuery.matches
-  }
```

Comment:
can we prevent the switch from "animating" every time we load a page in dark mode please? this makes the rendering time quite lengthy . I rather only have the animation when somethiing changes by the user, but not on every page.
This is also one of the reasons why the cypress tests are failing.

Reply:
i tried but i am not able to resolve that flickering issue every time we navigate to different page, but about the cypress test its failing cause they are written as respect to radio button logic and i have updated to checkbox hence they are failing. so i am tweaking the test script a bit

Reply:
the flickering issue: might that have to do with how you load the initial state and how you render it ;-)?
Happy to review again when you have the tests fixed and the attributes `th:attr="data-cy='dark-mode-<widgettype>'"` back again.

Reply:
cypress seems to work, i guess there are 2 things left:
- In the console of the browser, I now get an error: "Error: <path> attribute d: Expected number, "…7.18492 13.0678 ..."."
- it seems that when we set it to "checked" in the dom it only animates when we want to go to light mode. @mikewoudenberg would you know what to do here?